esti/migrate_test.go

```diff
@@ -0,0 +1,128 @@
+	"github.com/treeverse/lakefs/pkg/api"
+)
+
+func TestMigrate(t *testing.T) {
+	postMigrate := viper.GetViper().GetBool("post_migrate")
+
+	if postMigrate {
+		postMigrateTests(t)
+	} else {
+		preMigrateTests(t)
+	}
+}
+
+func preMigrateTests(t *testing.T) {
+	// all pre tests execution
+	t.Run("TestPreMigrateMultipart", testPreMigrateMultipart)
+
+	saveStateInLakeFS(t)
+}
+
+func postMigrateTests(t *testing.T) {
+	readStateFromLakeFS(t)
+
+	// all post tests execution
+	t.Run("TestPostMigrateMultipart", testPostMigrateMultipart)
+}
+
+func saveStateInLakeFS(t *testing.T) {
+	// write the state file
+	stateBytes, err := json.Marshal(&state)
+	require.NoError(t, err, "marshal state")
+
+	ctx := context.Background()
+	_ = createRepositoryByName(context.Background(), t, migrateStateRepoName)
+	resp, err := uploadContent(ctx, migrateStateRepoName, "main", migrateStateObjectPath, string(stateBytes))
+	require.NoError(t, err, "writing state file")
+	require.Equal(t, http.StatusCreated, resp.StatusCode())
+}
+
+func readStateFromLakeFS(t *testing.T) {
+	// read the state file
+	ctx := context.Background()
+	resp, err := client.GetObjectWithResponse(ctx, migrateStateRepoName, "main", &api.GetObjectParams{Path: migrateStateObjectPath})
+	require.NoError(t, err, "reading state file")
+	require.Equal(t, http.StatusOK, resp.StatusCode())
+
+	err = json.Unmarshal(resp.Body, &state)
+	require.NoError(t, err, "unmarshal state from response")
+}
+
+// state to be used
+var state migrateTestState
+
+type migrateTestState struct {
+	Multipart multipartState
+}
+
+type multipartState struct {
+	Repo           string                         `json:"repo"`
+	Info           s3.CreateMultipartUploadOutput `json:"info"`
+	CompletedParts []*s3.CompletedPart            `json:"completed_parts"`
+	Content        string                         `json:"state"`
+}
+
+const (
+	migrateMultipartFile     = "multipart_file"
+	migrateMultipartFilepath = mainBranch + "/" + migrateMultipartFile
+	migrateStateRepoName     = "migrate"
+	migrateStateObjectPath   = "state.json"
+)
+
+func testPreMigrateMultipart(t *testing.T) {
+	_, logger, repo := setupTest(t)
+
+	input := &s3.CreateMultipartUploadInput{
+		Bucket: aws.String(repo),
+		Key:    aws.String(migrateMultipartFilepath),
+	}
+
+	resp, err := svc.CreateMultipartUpload(input)
+	require.NoError(t, err, "failed to create multipart upload")
+	logger.Info("Created multipart upload request")
+
+	parts := make([][]byte, multipartNumberOfParts)
+	var partsConcat []byte
+	for i := 0; i < multipartNumberOfParts; i++ {
+		parts[i] = randstr.Bytes(multipartPartSize + i)
+		partsConcat = append(partsConcat, parts[i]...)
+	}
+
+	completedParts := uploadMultipartParts(t, logger, resp, parts)
+
+	state.Multipart.Repo = repo
+	state.Multipart.Info = *resp
+	state.Multipart.CompletedParts = completedParts
+	state.Multipart.Content = base64.StdEncoding.EncodeToString(partsConcat)
+}
+
+func testPostMigrateMultipart(t *testing.T) {
```

Comment:
I would arrange all the test functions together and order them hierarchically and put the variables declaration at the top of the module